runtime/src/iree/vm/ref_cc.h

```diff
@@ -354,6 +354,15 @@ class ref {
   mutable iree_vm_ref_t ref_;
 };
 
+// Constructs an object of type T and wraps it in a reference.
+//
+// Usage:
+//   ref<MyType> a = make_ref<MyType>(...);
+template <typename T, typename... Args>
+inline ref<T> make_ref(Args... args) {
+  return ref<T>(new T(std::forward<Args>(args)...));
```

Comment:
Since `args` are passed by value, I don't think this `forward` does anything. You can either add `&&` to `Args` (`Args&& ... args`) or use `std::move` if you prefer to keep the pass by value semantics.